CandyMarket/CandyMarket/DataAccess/CandyRepo.cs

```diff
@@ -181,6 +197,33 @@ from UserCandies
             }
         }
 
+        // ConsumeRandomCandy: finds UserCandyId for oldest piece of a random candy by flavor and calls EatCandy
+        public UserCandyDetailed ConsumeRandomCandy(string flavor, int userId)
+        {
+            var sql = @"select top 1 with ties UserCandyId
+                        from UserCandies uc
+                            join Candies c
+                            on uc.candyId = c.candyId
+                        where isConsumed = 0
+                        and flavorCategory = @Flavor
+                        and userId = @UserId
+                        order by dateReceived";
+
+            using (var db = new SqlConnection(connectionString))
+            {
+                var parameters = new
+                {
+                    Flavor = flavor,
+                    UserId = userId
+                };
+                var CandyOptions = db.Query<int>(sql, parameters);
+                var RNG = new Random().Next(0, CandyOptions.Count());
+                var CandyToConsume = CandyOptions.ElementAtOrDefault(RNG);
+                var eatenCandy = EatCandy(CandyToConsume);
```

Comment:
The Eat Candy Method needs the usercandyId to consume the candy.  Change CandyToConsume to "CandyToConsume.UserCandyId"

Reply:
CandyToConsume already is a UserCandyId. CandyOptions is an IEnumberable collection of UserCandyIds from which we selected a random index